source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -102,20 +101,45 @@ void Filter::onComplete(Filters::Common::ExtAuthz::CheckStatus status) {
     break;
   }
 
+  ENVOY_STREAM_LOG(trace, "Ext_authz received status code {}", *callbacks_,
+                   enumToInt(response->status_code));
+
   // We fail open/fail close based of filter config
   // if there is an error contacting the service.
-  if (status == CheckStatus::Denied ||
-      (status == CheckStatus::Error && !config_->failureModeAllow())) {
-    callbacks_->sendLocalReply(Http::Code::Forbidden, "", nullptr);
+  if (response->status == CheckStatus::Denied ||
+      (response->status == CheckStatus::Error && !config_->failureModeAllow())) {
+    ENVOY_STREAM_LOG(debug, "Ext_authz rejected the request", *callbacks_);
+
+    callbacks_->sendLocalReply(
+        response->status_code, response->body,
+        [& authz_headers = response->headers,
```

Comment:
Nit: weird spacing here, surprised this isn't failing check_format.

Reply:
It was done by `./ci/run_envoy_docker.sh './ci/do_ci.sh fix_format'`. I can try to fix it.

Reply:
If that's what clang-format says, I guess we should go with that unless there is a simple tweak to make it less ugly..

Reply:
No luck with this one. Remains ugly :(

Reply:
No luck with this one. Remains ugly :(